### PR TITLE
nspawn: fix overlay with automatic temporary tree

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/nspawn/nspawn-mount.c [hunk 1]
```diff
@@ -102,6 +102,32 @@ static char *resolve_source_path(const char *dest, const char *source) {
         return strdup(source);
 }
 
+static int allocate_temporary_source(CustomMount *m) {
+        _cleanup_free_ char *t = NULL, *s = NULL;
```
Comment (mrc0mmand) [inline src/nspawn/nspawn-mount.c]:
```
    archlinux_systemd_ci: ../src/nspawn/nspawn-mount.c:106:30: error: unused variable 't' [-Werror,-Wunused-variable]
    archlinux_systemd_ci:         _cleanup_free_ char *t = NULL, *s = NULL;
    archlinux_systemd_ci:                              ^
    archlinux_systemd_ci: ../src/nspawn/nspawn-mount.c:106:41: error: unused variable 's' [-Werror,-Wunused-variable]
    archlinux_systemd_ci:         _cleanup_free_ char *t = NULL, *s = NULL;
    archlinux_systemd_ci:                                         ^
    archlinux_systemd_ci: 2 errors generated.
```

Comment (poettering) [reply src/nspawn/nspawn-mount.c]: indeed! will fix

**Comment by yuwata** [comment]
LGTM.

**Comment by poettering** [comment]
microsoft CI is borked. All others passed. merging.

**Pull request merged**